game/scripts/vscripts/heroes/tusk/snowball.lua

```diff
@@ -0,0 +1,362 @@
+    [MODIFIER_STATE_ROOTED] = IsServer(),
+    [MODIFIER_STATE_MAGIC_IMMUNE] = true,
+    --[MODIFIER_STATE_INVULNERABLE] = true, -- These state make allied orders impossible
+    --[MODIFIER_STATE_OUT_OF_GAME] = true,
+    [MODIFIER_STATE_NO_HEALTH_BAR] = true,
+    [MODIFIER_STATE_UNSELECTABLE] = false,
+  }
+end
+
+---@return
+---@param keys table
+  function modifier_tusk_snowball_host:OnOrder(keys)
+  local caster = self:GetCaster()
+  -- Only act on normal or attack order, this also ensures that there is a target
+  if keys.order_type ~= DOTA_UNIT_ORDER_MOVE_TO_TARGET and keys.order_type ~= DOTA_UNIT_ORDER_ATTACK_TARGET then
+    return
+  end
+
+  if keys.unit == keys.target then return end
+
+  -- Check if either the issuer or the target is the caster
+  if keys.target ~= caster and keys.unit ~= caster then return end
+
+  -- Only allied units can be caught
+  if keys.target:GetTeamNumber() ~= caster:GetTeamNumber() then return end
+  -- Filter out units, illusions do count
+  if not keys.target:IsHero() then return end
+  if keys.unit ~= caster and PlayerResource:IsDisableHelpSetForPlayerID(keys.target:GetPlayerOwnerID(),caster:GetPlayerOwnerID()) then return end
+
+  -- Check if the unit is in range for the grab
+  if (keys.target:GetAbsOrigin() - caster:GetAbsOrigin()):Length2D() >= self.snowball_grab_radius then return end
+
+  local unit = keys.target
+  if unit == caster then
+    unit = keys.unit
+  end
+
+  -- No more reasons to reject
+  self:GetAbility().unitInSnowball[unit] = true
+
+  unit:AddNewModifier(caster,self:GetAbility(),"modifier_tusk_snowball_guest",{duration = 3})
+end
+
+
+
+-- The modifier that handles everything for allies
+LinkLuaModifier("modifier_tusk_snowball_guest","heroes/tusk/snowball.lua",LUA_MODIFIER_MOTION_NONE)
+---@class modifier_tusk_snowball_guest : CDOTA_Modifier_Lua
+modifier_tusk_snowball_guest = class({})
+---@override
+function modifier_tusk_snowball_guest:IsHidden() return true end
+---@override
+function modifier_tusk_snowball_guest:IsPurgable() return false end
+---@override
+function modifier_tusk_snowball_guest:IsPurgeException() return false end
+---@override
+function modifier_tusk_snowball_guest:IsDebuff() return false end
+
+---@override
+function modifier_tusk_snowball_guest:OnCreated()
+  if IsServer() then -- Nothing needs to be done on the client
+    --Show some effect
+    local particle = ParticleManager:CreateParticle("particles/units/heroes/hero_tusk/tusk_snowball_load.vpcf", PATTACH_ABSORIGIN_FOLLOW, self:GetParent())
+    ParticleManager:SetParticleControl(particle, 0, self:GetParent():GetAbsOrigin())
+    ParticleManager:ReleaseParticleIndex(particle)
+
+    self:GetParent():EmitSound("Hero_Tusk.Snowball.Ally")
+    self:GetParent():AddNoDraw()
+  end
+end
+
+---@override
+function modifier_tusk_snowball_guest:OnDestroy()
+  if IsServer() then -- Nothing needs to be done on the client
+    self:GetParent():RemoveNoDraw()
```

Comment:
Maybe add a FindClearSpaceForUnit here

Reply:
ResolveNPCPositions is used when the projectile hits